command/agent/config.go

```diff
@@ -1055,6 +1088,9 @@ func (a *ServerConfig) Merge(b *ServerConfig) *ServerConfig {
 	if b.EncryptKey != "" {
 		result.EncryptKey = b.EncryptKey
 	}
+	if b.ServerJoin != nil {
+		result.ServerJoin = b.ServerJoin
```

Comment:
This is overriding not merging. You need a merge method on the ServerJoin object